### ISSUE TITLE
[dependency-injection] does not autoconfigure my form extension when another form extension is configured manually in services.yaml

### ISSUE THREAD

**Issue by TomaszGasior** [symfony/symfony#35916]
**Symfony version(s) affected**: 4.4.5

**Description**  
After updating `symfony/dependency-injection` from 4.4.4 to 4.4.5 my one form extension is not autoconfigured (tagged) by Symfony framework. If I remove manual configuration of another form extension from `services.yaml`, the first one is properly autoconfigured (tagged).

**How to reproduce**  
1. Clone my project: https://github.com/TomaszGasior/RadioLista-v3. Chechout 44f1ef41f1dd115f3405730bc8eaee512d10d325 commit where all dependencies except `symfony/dependency-injection` are up to date at the moment where I write this text.
2. `bin/phpunit` -> tests should pass
3. Update symfony/dependency-injection to 4.4.5.
4. `bin/phpunit` -> Tests related to adding, removing and editing radio stations should fail. Twig templates of these pages rely on variable from RadioStationEditExtension form extension which is not autoconfigured.
5. Remove manual configuration of `App\Form\Extension\ValidationFlashErrorExtension` from `services.yaml`. (My app contains two form extensions. ValidationFlashErrorExtension needs to be set up manually to get special priority.)
6. `bin/phpunit` -> tests should pass

**Comment by nutama** [comment]
I noticed the same behaviour with the ArgumentValueResolver, but this issue can happen for every service that is autoconfigured. This seems to be related to #35342 .

Before 4.4.5, tagged services are only processed once by the [PriorityTaggedServiceTrait](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/DependencyInjection/Compiler/PriorityTaggedServiceTrait.php). Since 4.4.5, services are processes multiple times, based on the number of tags.

This works fine when:
* autoconfigure is disabled and you manually tag your services
* autoconfigure is enabled and you do not manually tag your services with a priority
* autoconfigure is enabled and you do tag all your services with a priority

This can not work when:
* autoconfigure is enabled and you mixed manually tagged services with an priority and depend on autoconfiguring

Autoconfigure automaticly add tags to a service.
When you manually define a service that is also autoconfigure, multiple tags are added to the definition.
It can work, cause it depends on the order (abc.. asc) when services are loaded.

For example:
```yaml
services:
    _defaults:
        autowire: true
        autoconfigure: true

    App\ArgumentResolver\AppArgumentValueResolver:
        tags:
            - { name: controller.argument_value_resolver, priority: 150 }
```

Then you will a service with the following attributes:
```
array(2) {
  [0] =>
  array(1) {
    'priority' =>
    int(150)
  }
  [1] =>
  array(0) {
  }
}
```

The current code in the PriorityTaggedServiceTrait will overwrite services, when there is already a service with priority 0.

The current unittest does not take this case into account, and does not test scenarios with multiple attributes.

Change in test to trigger the bug:

```patch
@@ -21,25 +21,25 @@ class PriorityTaggedServiceTraitTest extends TestCase
     public function testThatCacheWarmersAreProcessedInPriorityOrder()
     {
         $services = [
-            'my_service1' => ['my_custom_tag' => ['priority' => 100]],
-            'my_service2' => ['my_custom_tag' => ['priority' => 200]],
-            'my_service3' => ['my_custom_tag' => ['priority' => -501]],
-            'my_service4' => ['my_custom_tag' => []],
-            'my_service5' => ['my_custom_tag' => ['priority' => -1]],
-            'my_service6' => ['my_custom_tag' => ['priority' => -500]],
-            'my_service7' => ['my_custom_tag' => ['priority' => -499]],
-            'my_service8' => ['my_custom_tag' => ['priority' => 1]],
-            'my_service9' => ['my_custom_tag' => ['priority' => -2]],
-            'my_service10' => ['my_custom_tag' => ['priority' => -1000]],
-            'my_service11' => ['my_custom_tag' => ['priority' => -1001]],
-            'my_service12' => ['my_custom_tag' => ['priority' => -1002]],
-            'my_service13' => ['my_custom_tag' => ['priority' => -1003]],
-            'my_service14' => ['my_custom_tag' => ['priority' => -1000]],
-            'my_service15' => ['my_custom_tag' => ['priority' => 1]],
-            'my_service16' => ['my_custom_tag' => ['priority' => -1]],
-            'my_service17' => ['my_custom_tag' => ['priority' => 200]],
-            'my_service18' => ['my_custom_tag' => ['priority' => 100]],
-            'my_service19' => ['my_custom_tag' => []],
+            'my_service1' => ['my_custom_tag' => [['priority' => 100]]],
+            'my_service2' => ['my_custom_tag' => [['priority' => 200]]],
+            'my_service3' => ['my_custom_tag' => [['priority' => -501]]],
+            'my_service4' => ['my_custom_tag' => [[]]],
+            'my_service5' => ['my_custom_tag' => [['priority' => -1]]],
+            'my_service6' => ['my_custom_tag' => [['priority' => -500]]],
+            'my_service7' => ['my_custom_tag' => [['priority' => -499]]],
+            'my_service8' => ['my_custom_tag' => [['priority' => 1]]],
+            'my_service9' => ['my_custom_tag' => [['priority' => -2]]],
+            'my_service10' => ['my_custom_tag' => [['priority' => -1000]]],
+            'my_service11' => ['my_custom_tag' => [['priority' => -1001]]],
+            'my_service12' => ['my_custom_tag' => [['priority' => -1002]]],
+            'my_service13' => ['my_custom_tag' => [['priority' => -1003]]],
+            'my_service14' => ['my_custom_tag' => [['priority' => -1000]]],
+            'my_service15' => ['my_custom_tag' => [['priority' => 1]]],
+            'my_service16' => ['my_custom_tag' => [['priority' => -1]]],
+            'my_service17' => ['my_custom_tag' => [['priority' => 200]]],
+            'my_service18' => ['my_custom_tag' => [['priority' => 100], []]],
+            'my_service19' => ['my_custom_tag' => [[]]],
         ];
 
         $container = new ContainerBuilder();
@@ -48,7 +48,9 @@ class PriorityTaggedServiceTraitTest extends TestCase
             $definition = $container->register($id);
 
             foreach ($tags as $name => $attributes) {
-                $definition->addTag($name, $attributes);
+                foreach ($attributes as $attribute) {
+                    $definition->addTag($name, $attribute);
+                }
             }
         }
```

As result: my_service18 will now overwrite / remove my_service4

**Comment by apfelbox** [comment]
This is a breaking bug in Symfony 4.4.5 / 5.0.5, as some services are just randomly disappearing

Fix for now: pin the version to 4.4.4 / 5.0.4

FYI: @nicolas-grekas

**Comment by alex-dev** [comment]
I'll check it. Seems like I caused the regression.

**Comment by nicolas-grekas** [comment]
I just fixed this: #35957

**Comment by TomaszGasior** [comment]
@nicolas-grekas I confirm that your PR fixes my original issue.
